miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -584,93 +585,91 @@ HotTarget.sample = (function() {
-                  return sampleClass.sampleCategory == "Aliquot";
-                })) {
-                  targets.push({
-                    name: "Library",
-                    action: function(replicates) {
-                      window.location = "/miso/library/bulk/propagate?" + jQuery.param({
-                        ids: idsString,
-                        replicates: replicates
-                      });
-                    }
+      }, {
+        name: "Propagate",
+        action: function(samples) {
+          HotUtils.warnIfConsentRevoked(samples, function() {
+            var idsString = samples.map(Utils.array.getId).join(",");
+            var classes = getSampleClasses(samples);
+
+            // In the case of plain samples, this will be empty, which is fine.
+            var targets = getChildSampleClasses(classes).sort(Utils.sorting.sampleClassComparator).map(function(sampleClass) {
+
+              return {
+                name: sampleClass.alias,
+                action: function(replicates) {
+                  window.location = "/miso/sample/bulk/propagate?" + jQuery.param({
+                    parentIds: idsString,
+                    replicates: replicates,
+                    sampleClassId: sampleClass.id
                   });
                 }
-
-                if (targets.length == 0) {
-                  alert("No propagation is possible from the samples.");
-                  return;
+              };
+
+            });
+            if (!Constants.isDetailedSample || classes.every(function(sampleClass) {
+              return sampleClass.sampleCategory == "Aliquot";
+            })) {
+              targets.push({
+                name: "Library",
+                action: function(replicates) {
+                  window.location = "/miso/library/bulk/propagate?" + jQuery.param({
+                    ids: idsString,
+                    replicates: replicates
+                  });
                 }
-
-                Utils.showDialog(targets.length > 1 ? 'Propagate Samples' : ('Propagate to ' + targets[0].name), 'Propagate', [{
-                  property: 'replicates',
-                  type: 'int',
-                  label: 'Replicates',
-                  value: 1
-                }, targets.length > 1 ? {
-                  property: 'target',
-                  type: 'select',
-                  label: 'To',
-                  values: targets,
-                  getLabel: Utils.array.getName
-                } : null].filter(function(x) {
-                  return !!x;
-                }), function(result) {
-                  (result.target || targets[0]).action(result.replicates);
-                });
               });
             }
-          }, HotUtils.printAction('sample'), HotUtils.spreadsheetAction('/miso/rest/sample/spreadsheet', Constants.sampleSpreadsheets),
+
+            if (targets.length == 0) {
+              alert("No propagation is possible from the samples.");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this usage of alert(...). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS1442)